tests/e2e/squid_deployment_test.go

```diff
@@ -253,16 +238,6 @@ var _ = Describe("Squid Helm Chart Deployment", func() {
 						Expect(*squidContainer.SecurityContext.RunAsNonRoot).To(BeTrue())
 					}
 				}
-
-				// Find squid-exporter container
-				var exporterContainer *corev1.Container
-				for i := range pod.Spec.Containers {
-					if pod.Spec.Containers[i].Name == "squid-exporter" {
-						exporterContainer = &pod.Spec.Containers[i]
-						break
-					}
-				}
-				Expect(exporterContainer).NotTo(BeNil(), "squid-exporter container should exist")
```

Comment:
Shouldn't the squid-exporter container exist still? We probably want to keep this check. However, let's do away with the loop and use the same approach as above.